util/otel/meter.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package otel
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	promex "go.opentelemetry.io/otel/exporters/prometheus"
+	otel "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/noop"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configtelemetry"
+)
+
+func MeterProvider(reg prometheus.Registerer) (otel.MeterProvider, error) {
```

Comment:
To make this initial PR smaller, would it be ok if we tackle telemetry in a follow up PR? Or does the processor doesn't work at all without a registerer?

Reply:
We need to pass the `TelemetrySettings`, or deltatocumulative will panic.
This has to include a meter.

I can change this to always pass the `noop` meter, but this will only save us half this file.
Will do if you consider this worth it!

Reply:
This whole file could be only this:

```go
package otel

import (
	"github.com/prometheus/client_golang/prometheus"

	otel "go.opentelemetry.io/otel/metric"
	"go.opentelemetry.io/otel/metric/noop"

	"go.opentelemetry.io/collector/component"
	"go.opentelemetry.io/collector/config/configtelemetry"
)

func TelemetrySettings(_ prometheus.Registerer) (component.TelemetrySettings, error) {
	mp := noop.NewMeterProvider()
	return component.TelemetrySettings{
		MeterProvider: mp,
		LeveledMeterProvider: func(_ configtelemetry.Level) otel.MeterProvider {
			return mp
		},
	}, nil
}
```